synapse/handlers/message.py

```diff
@@ -385,7 +385,7 @@ def __init__(self, hs: "HomeServer"):
         self._events_shard_config = self.config.worker.events_shard_config
         self._instance_name = hs.get_instance_name()
 
-        self.room_invite_state_types = self.hs.config.room_invite_state_types
+        self.room_invite_state_types = self.hs.config.api.room_prejoin_state
```

Comment:
Is this preferred over `self.hs.config.room_prejoin_state` for performance reasons?

Reply:
in my ideal world, we'd get rid of all the direct accesses to `config.foo` fields, and remove the magic in `RootConfig` that makes it work.

In the meantime, I'm working to reduce instances of it, one by one.